### ISSUE TITLE
Build Fails for iOS

### ISSUE THREAD

**Issue by tomwanzek** [RevenueCat/react-native-purchases#257]
- [X] I have updated Purchases SDK to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/RevenueCat/react-native-purchases/blob/develop/CONTRIBUTING.md)
- [X] I have read the [Help Center](https://support.revenuecat.com/hc/en-us)
- [X] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [X] I have searched for [existing Github issues](https://github.com/RevenueCat/react-native-purchases/issues)

**Describe the bug**

1. Environment
   1. Platform: iOS
   2. SDK version: 4.1.1 and 4.1.2
   3. OS version:
   4. XCode/Android Studio version: XCode 12.5
   5. React Native version: 0.63.4
   6. SDK installation (CocoaPods + version or manual): pod v1.10.1
   7. How widespread is the issue. Percentage of devices affected: See additional context below 
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue

With version 4.1.1 of the SDK the build error we are experiencing comes out in Circle CI as:

>The following build commands failed:
	CompileSwiftSources normal armv7 com.apple.xcode.tools.swift.compiler
	CompileSwift normal armv7 /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/PurchasesHybridCommon/PurchasesHybridCommon.swift
	CompileSwift normal arm64 /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/PurchasesHybridCommon/PurchasesHybridCommon.swift

With version 4.1.2 of the SDK the build error seems to shift a little:

The following build commands failed:
	CompileSwiftSources normal armv7 com.apple.xcode.tools.swift.compiler
	CompileSwift normal armv7 /Users/distiller/someorg/someproduct/someproduct-app/node_modules/react-native-purchases/ios/PurchasesPlugin.swift
	CompileSwift normal arm64 /Users/distiller/someorg/someproduct/someproduct-app/node_modules/react-native-purchases/ios/PurchasesPlugin.swift

3. Steps to reproduce, with a description of expected vs. actual behavior

See additional context below

4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)

We are wondering, whether the hopefully quickly released fix to https://github.com/RevenueCat/react-native-purchases/issues/255 addresses this issue in a stable manner.

**Additional context**

The build failure is breaking the CI workflow for one of our app versions, however, it is intermittent in the following sense.

We were ale to build and deploy in CI with version 4.1.1 of the SDK, a couple of days ago after running a Staging deploy the build error above started to occur suddenly even though we had not changed any Purchases SDK related dependencies. Simply re-running from failed or re-running from start did not resolve the issue.

We made another deploy with a minor change to the CI config to pin the XCode version to 12.5. This deployment succeeded.
However, the next deployment failed again with the same error as above for SDK version 4.1.1.

After upgrading the SDK to version 4.1.2 the error has shifted from `PurchasesHybridCommon` to `PurchasesPlugin`, but essentially stops us cold.

**Comment by tomwanzek** [comment]
Here is a little more of the CI build output:

```
/Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Public/React-Core/React/RCTJSInvokerModule.h:14:64: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified)

@property (nonatomic, copy, nonnull) void (^invokeJS)(NSString *module, NSString *method, NSArray *args);
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/node_modules/react-native/React/Modules/RCTEventEmitter.h:17:39: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified)

@property (nonatomic, weak) RCTBridge *bridge;
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/node_modules/react-native/React/Modules/RCTEventEmitter.h:25:24: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified)

- (NSArray<NSString *> *)supportedEvents;
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/node_modules/react-native/React/Modules/RCTEventEmitter.h:31:37: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified)

- (void)sendEventWithName:(NSString *)name body:(id)body;
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/node_modules/react-native/React/Modules/RCTEventEmitter.h:31:50: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified)

- (void)sendEventWithName:(NSString *)name body:(id)body;
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/node_modules/react-native/React/Modules/RCTEventEmitter.h:41:31: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified)

- (void)addListener:(NSString *)eventName;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:60:1: duplicate interface definition for class 'RCPurchases'

@interface RCPurchases : NSObject
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:65:43: property has a previous declaration

@property (class, nonatomic, assign) BOOL automaticAppleSearchAdsAttributionCollection;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:70:43: property has a previous declaration

@property (class, nonatomic, assign) BOOL debugLogsEnabled;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:75:53: property has a previous declaration

@property (class, nonatomic, copy, nullable) NSURL *proxyURL;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:81:43: property has a previous declaration

@property (class, nonatomic, assign) BOOL forceUniversalAppStore;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:87:43: property has a previous declaration

@property (class, nonatomic, assign) BOOL simulatesAskToBuyInSandbox API_AVAILABLE(ios(8.0), macos(10.14), watchos(6.2), macCatalyst(13.0), tvos(9.0));
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:156:53: property has a previous declaration

@property (class, nonatomic, readonly) RCPurchases *sharedPurchases;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:164:28: property has a previous declaration

@property (nonatomic) BOOL allowSharingAppStoreAccount;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:167:28: property has a previous declaration

@property (nonatomic) BOOL finishTransactions;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:173:63: property has a previous declaration

@property (nonatomic, weak, nullable) id<RCPurchasesDelegate> delegate;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:178:43: property has a previous declaration

@property (nonatomic, readonly) NSString *appUserID;
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:181:38: property has a previous declaration

@property (nonatomic, readonly) BOOL isAnonymous;
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/Purchases/RCPurchases.h:558:11: duplicate protocol definition of 'RCPurchasesDelegate' is ignored

@protocol RCPurchasesDelegate <NSObject>
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCPurchases+HybridAdditions.h:7:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCPurchases+HybridAdditions.h:12:12: definition of 'RCPurchases' must be imported from module 'Purchases.RCPurchases' before it is required

@interface RCPurchases (HybridAdditions)
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCCommonFunctionality.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCEntitlementInfo+HybridAdditions.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCEntitlementInfo+HybridAdditions.h:8:12: declaration of 'RCEntitlementInfo' must be imported from module 'Purchases.RCEntitlementInfos' before it is required

@interface RCEntitlementInfo (HybridAdditions)
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCEntitlementInfo+HybridAdditions.h:8:12: definition of 'RCEntitlementInfo' must be imported from module 'Purchases.RCEntitlementInfo' before it is required

@interface RCEntitlementInfo (HybridAdditions)
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCEntitlementInfos+HybridAdditions.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCErrorContainer.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCOffering+HybridAdditions.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCOffering+HybridAdditions.h:8:12: definition of 'RCOffering' must be imported from module 'Purchases.RCOffering' before it is required

@interface RCOffering (HybridAdditions)
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCOfferings+HybridAdditions.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCOfferings+HybridAdditions.h:8:12: definition of 'RCOfferings' must be imported from module 'Purchases.RCOfferings' before it is required

@interface RCOfferings (HybridAdditions)
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCPackage+HybridAdditions.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCPackage+HybridAdditions.h:8:12: definition of 'RCPackage' must be imported from module 'Purchases.RCPackage' before it is required

@interface RCPackage (HybridAdditions)
        ^



⚠️  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCPurchaserInfo+HybridAdditions.h:6:2: missing submodule 'Purchases.Purchases'

#import <Purchases/Purchases.h>
        ^



❌  /Users/distiller/someorg/someproduct/someproduct-app/ios/Pods/Headers/Private/PurchasesHybridCommon/RCPurchaserInfo+HybridAdditions.h:8:12: definition of 'RCPurchaserInfo' must be imported from module 'Purchases.RCPurchaserInfo' before it is required

@interface RCPurchaserInfo (HybridAdditions)
        ^


▸ Compiling PurchasesPlugin.swift
```

**Comment by beylmk** [comment]
@tomwanzek could you please confirm whether this was fixed by release 4.14?

**Comment by tomwanzek** [comment]
@beylmk I will close this issue, while we have not yet upgraded to 4.14 yet, we managed to seemingly address this issue through some changes to our CI cache. If it re-appears, I will update here or re-open. Cheers 😄
